### ISSUE TITLE
Crash when accessing jump menu

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#998]
Stack trace from Firebase Crash. Happens across API levels.

```
Exception java.lang.RuntimeException: Unable to start activity ComponentInfo{org.odk.collect.android/org.odk.collect.android.activities.FormHierarchyActivity}: java.lang.NullPointerException
android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2436)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2498)
android.app.ActivityThread.access$900 (ActivityThread.java:179)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1324)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:146)
android.app.ActivityThread.main (ActivityThread.java:5641)
java.lang.reflect.Method.invokeNative (Method.java)
java.lang.reflect.Method.invoke (Method.java:515)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1288)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1104)
dalvik.system.NativeStart.main (NativeStart.java)

Caused by java.lang.NullPointerException:
org.odk.collect.android.activities.FormHierarchyActivity.onCreate (FormHierarchyActivity.java:74)
android.app.Activity.performCreate (Activity.java:5484)
android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1093)
android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2400)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2498)
android.app.ActivityThread.access$900 (ActivityThread.java:179)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1324)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:146)
android.app.ActivityThread.main (ActivityThread.java:5641)
java.lang.reflect.Method.invokeNative (Method.java)
java.lang.reflect.Method.invoke (Method.java:515)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1288)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1104)
dalvik.system.NativeStart.main (NativeStart.java)
```

**Comment by grzesiek2010** [comment]
@lognaturel I'm not able to reproduce the issue. Looks like [Collect.getInstance().getFormController();](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java#L71) returns null.

Theoretically it would be possible if user click on "Go to prompt" icon during form loading but it's not possible as the UI is blocked during form loading.

FormController is set [here](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java#L2474) so as I said above theoretically it would be possible but UI is blocked so we can't do that.

**Comment by grzesiek2010** [comment]
@lognaturel 
<s>I'm not able to reproduce the issue. Looks like Collect.getInstance().getFormController(); returns null.

Theoretically it would be possible if user clicks on "Go to prompt" icon during form loading but it's not possible as the UI is blocked during form loading.

FormController is set here so as I said above theoretically it would be possible but UI is blocked so we can't do that.

I've also tried some strange scenarios with "Don't keep activities" option without success as well.

The only one explanation is that someone is able to click on "Go to prompt" icon after dismissDialog but before https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java#L2474 :)
Maybe we should dismiss that dialog at the end of the method?

Is someone fast enough? :)</s>

Got it! It takes place when we close the app (when the FormHierarchyActivity is opened) using a physical button and leave the app for a long time (in my case it's at least 30 minutes), then if we open the app again the app crashes.

I've added a small trick to solve that.
